src/screen/dashboard.rs

```diff
@@ -1488,14 +1488,25 @@ impl Dashboard {
                     &message_reference_types,
                 );
 
-                clients.send_chathistory_request(
-                    server,
-                    ChatHistorySubcommand::Before(
-                        target.clone(),
-                        first_can_reference,
-                        clients.get_server_chathistory_limit(server),
-                    ),
-                );
+                if matches!(first_can_reference, MessageReference::None) {
+                    clients.send_chathistory_request(
+                        server,
+                        ChatHistorySubcommand::Latest(
+                            target.clone(),
+                            first_can_reference,
+                            clients.get_server_chathistory_limit(server),
+                        ),
+                    );
+                } else {
+                    clients.send_chathistory_request(
+                        server,
+                        ChatHistorySubcommand::Before(
+                            target.clone(),
+                            first_can_reference,
+                            clients.get_server_chathistory_limit(server),
+                        ),
+                    );
+                }
```

Comment:
Nit: i think i would if/else the subcommand, and leave a single `send_chathistory_request` for readability.

```rust
let subcommand = if matches!(first_can_reference, MessageReference::None) {
    ChatHistorySubcommand::Latest(
        target.clone(),
        first_can_reference,
        clients.get_server_chathistory_limit(server),
    )
} else {
    ChatHistorySubcommand::Before(
        target.clone(),
        first_can_reference,
        clients.get_server_chathistory_limit(server),
    )
};

clients.send_chathistory_request(server, subcommand);
```